### PR TITLE
Fixing Rename Refactorings

### PR DESCRIPTION
**Pull request by dantleech**
- Rename WorkspaceRenameEdits => WorkspaceOperations
- Removes dead variable
- Use `merge` to accumulate changes to the operations
- Only convert once
- Use IteratorAggregate instead of extending ArrayIterator

and finally makes the unit of operation the LocatedTextDocumentEdits and not the LocatedTextDocumentEditsMap

### CODE REVIEW

**Comment by przepompownia** [comment]
Looks fine at first glance but I get incomplete response (without text edits), for example

```lua
{
  documentChanges = { {
      kind = "rename",
      newUri = "file:///tmp/phpactor-dirty/lib/Name/Phpactor2.php",
      oldUri = "file:///tmp/phpactor-dirty/lib/Phpactor.php"
    } }
}
```

and haven't check why yet.

File: lib/Rename/Adapter/ClassMover/FileRenamer.php [hunk 1]
```diff
@@ -66,8 +66,8 @@ public function renameFile(TextDocumentUri $from, TextDocumentUri $to): Promise
                 }
             }
 
-            return new WorkspaceRenameEdits([
-                LocatedTextEditsMap::fromLocatedEdits($locatedEdits),
+            return new WorkspaceOperations([
+                LocatedTextEditsMap::fromLocatedEdits($locatedEdits)->toLocatedTextEdits(),
```
Comment (przepompownia) [inline lib/Rename/Adapter/ClassMover/FileRenamer.php]:
```suggestion
                ...LocatedTextEditsMap::fromLocatedEdits($locatedEdits)->toLocatedTextEdits(),
```
Missing unpack here.

File: lib/Rename/Model/FileRenamer/TestFileRenamer.php [hunk 2]
```diff
@@ -8,18 +8,18 @@
...
 
     public function __construct(
         private bool $throw = false,
-        ?WorkspaceRenameEdits $renameEdit = null,
+        ?WorkspaceOperations $renameEdit = null,
     ) {
-        $this->renameEdit = $renameEdit ?: new WorkspaceRenameEdits(array_filter([
+        $this->renameEdit = $renameEdit ?: new WorkspaceOperations(array_filter([
             LocatedTextEditsMap::create(),
         ]));
```
Comment (przepompownia) [inline lib/Rename/Model/FileRenamer/TestFileRenamer.php]:
```suggestion
        $this->renameEdit = $renameEdit ?: new WorkspaceOperations(
            LocatedTextEditsMap::create()->toLocatedTextEdits(),
        );
```

File: lib/Rename/Model/FileRenamer/TestFileRenamer.php [hunk 3]
```diff
@@ -8,18 +8,18 @@
...
 use Phpactor\Rename\Model\FileRenamer;
 use Phpactor\Rename\Model\LocatedTextEditsMap;
-use Phpactor\Rename\Model\WorkspaceRenameEdits;
+use Phpactor\Rename\Model\WorkspaceOperations;
 use Phpactor\TextDocument\TextDocumentUri;
 
 class TestFileRenamer implements FileRenamer
 {
-    private WorkspaceRenameEdits $renameEdit;
+    private WorkspaceOperations $renameEdit;
```
Comment (przepompownia) [inline lib/Rename/Model/FileRenamer/TestFileRenamer.php:16]: `renameEdit` could be renamed to `workspaceOperations`.

**Comment by dantleech** [comment]
feel free to merge and change :)

**Review by przepompownia — APPROVED** [review]

**Pull request merged**